src/Snackbar/index.tsx

```diff
@@ -0,0 +1,197 @@
+    action,
+    message,
+    type = MESSAGE_TYPES.INFO,
+    renderInPortal = false
+}) => {
+    const [portal, setPortal] = useState<HTMLElement | null>(null);
+    const [snackbarWrapperElement, setSnackbarWrapperElement] = useState<HTMLElement | null>(null);
+    const [update, setUpdate] = useState<number>(0);
+    const snackbarContentRef = useRef<HTMLDivElement | null>(null);
+
+    useEffect(() => {
+        toggleSnackbarOpen();
+        let hideDuration = autoHideDuration;
+        // allow user to disable autoHideDuration
+        if (hideDuration === null) {
+            return;
+        }
+        // allow user to default autoHideDuration to 4000ms
+        if (hideDuration === undefined) {
+            hideDuration = 4000;
+        }
+        const timer = setTimeout(() => {
+            toggleSnackbarOpen(() => {
+                if (onClose) {
+                    onClose('timeout');
+                }
+            });
+        }, hideDuration);
+        return () => {
+            clearTimeout(timer);
+        };
+    }, [open, autoHideDuration]);
+
+    useEffect(() => {
+        const bodyElementHTMLCollection = document.getElementsByTagName('body');
+        const bodyElement = bodyElementHTMLCollection.length > 0 ? bodyElementHTMLCollection.item(0) : null;
+        let observer: IntersectionObserver;
+        let portalElement: HTMLElement;
+
+        if (renderInPortal && bodyElement) {
+            portalElement = document.createElement('div');
+
+            if (!portal) {
+                bodyElement.appendChild(portalElement);
+            }
+
+            if (!portal && snackbarWrapperElement) {
+                observer = new IntersectionObserver(() => {
+                    setUpdate(new Date().getTime());
+                });
+                observer.observe(snackbarWrapperElement);
+                setPortal(portalElement);
+            }
+        }
+
+        return () => {
+            if (snackbarWrapperElement && observer) {
+                observer.unobserve(snackbarWrapperElement);
+            }
+
+            if (portalElement && bodyElement && bodyElement.contains(portalElement)) {
+                bodyElement.removeChild(portalElement);
+            }
+        };
+    }, [snackbarWrapperElement, renderInPortal]);
+
+    useEffect(() => {
+        if (snackbarWrapperElement && portal) {
+            generateAnchorOrigin(anchorOrigin, portal);
+            portal.style.position = 'fixed';
+            portal.style.zIndex = `${theme?.zIndex?.snackbar}`;
+            portal.className = snackbarWrapperElement.className;
+        }
+    }, [update]);
+
+    const popover = useMemo(() => {
+        return (
+            <>
+                {open && (
+                    <div className="content-snackbar-wrapper">
+                        <SnackbarContent
+                            id="snackbar-content"
+                            ref={snackbarContentRef}
+                            action={action}
```

Comment:
If the user doesn't pass an action, we should add a default action here for a close button to stay inline to the expected functionality we've seen in other snackbar components. Basically what you have in the Storybook Snackbar story.